module-2/routes/users.ts

```diff
@@ -0,0 +1,84 @@
+import express from 'express';
+import { v4 as uuid } from 'uuid';
+import User from '../models/User';
+import { Users } from '../db';
+import schema from '../schema';
+import { filterNotDeleted, getAutoSuggestUsers } from '../utils';
+
+const router = express.Router();
+
+router.get('/', (req, res) => {
+  res.send([...Users.values()].filter(filterNotDeleted));
+});
+
+router.get('/all', (req, res) => {
+  res.send([...Users.values()]);
+});
+
+const DEFAULT_LIMIT = 3;
+router.get('/autosuggest', (req, res) => {
+  const { login = '', limit = DEFAULT_LIMIT } = req.query;
+  res.send(getAutoSuggestUsers(Users)(String(login), Number(limit)));
+});
+
+router.get('/:id', (req, res) => {
+  const user = Users.get(req.params.id);
+  !user?.isDeleted
```

Comment:
Consider rewriting this logic to improve readability.
From a code style perspective `if (user && user.isDeleted)` check is more readable.
Moreover, such expressions can be hard to write and debug which can lead to unexpected behaviour.
![image](https://user-images.githubusercontent.com/25229823/91099321-c63d0b80-e66b-11ea-856c-059451fdabd3.png)

Reply:
In the image above, as far as I can see, if there's no such user in DB we still successfully send "undefined" as a 200 response.

Reply:
Good catch, indeed a buggy check.
Rewrote to `!user || user.isDeleted` because `user && user.isDeleted` would also lead to bug in case when `user === undefined`
![image](https://user-images.githubusercontent.com/11832280/91938306-73fb8a80-ecfc-11ea-8b0a-2dbce13b01ab.png)